core/src/main/java/org/primefaces/extensions/component/suneditor/SunEditor.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2011-2023 PrimeFaces Extensions
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.primefaces.extensions.component.suneditor;
+
+import java.util.Collection;
+import java.util.Locale;
+
+import javax.faces.application.ResourceDependency;
+import javax.faces.component.behavior.ClientBehaviorHolder;
+import javax.faces.context.FacesContext;
+
+import org.primefaces.component.api.AbstractPrimeHtmlInputTextArea;
+import org.primefaces.component.api.RTLAware;
+import org.primefaces.component.api.Widget;
+import org.primefaces.util.LangUtils;
+import org.primefaces.util.LocaleUtils;
+
+/**
+ * <code>SunEditor</code> component.
+ *
+ * @author Matthieu Valente
+ * @since 12.0.6
+ */
+
+@ResourceDependency(library = "primefaces", name = "jquery/jquery.js")
+@ResourceDependency(library = "primefaces", name = "jquery/jquery-plugins.js")
+@ResourceDependency(library = "primefaces", name = "core.js")
+@ResourceDependency(library = "primefaces-extensions", name = "primefaces-extensions.js")
+@ResourceDependency(library = "primefaces-extensions", name = "suneditor/suneditor.min.css")
+@ResourceDependency(library = "primefaces-extensions", name = "suneditor/suneditor.min.js")
+@ResourceDependency(library = "primefaces-extensions", name = "suneditor/suneditor.css")
+@ResourceDependency(library = "primefaces-extensions", name = "suneditor/suneditor.js")
```

Comment:
Do we want to combine the JS into one file?

Reply:
Yes this was my comment about not using a separate module `resources-suneditor` I can then combine these as a normal widget.